core/src/test/resources/org/apache/calcite/test/RelOptRulesTest.xml

```diff
@@ -3295,9 +3295,9 @@ LogicalProject(DEPTNO=[$7])
       <![CDATA[
 LogicalProject(EMPNO=[$0])
   LogicalProject(EMPNO=[$0], ENAME=[$1], JOB=[$2], MGR=[$3], HIREDATE=[$4], SAL=[$5], COMM=[$6], DEPTNO=[$7], SLACKER=[$8], DEPTNO0=[$9], NAME=[$10])
-    LogicalJoin(condition=[true], joinType=[left])
+    LogicalCorrelate(correlation=[$cor0], joinType=[left], requiredColumns=[{7}])
       LogicalTableScan(table=[[CATALOG, SALES, EMP]])
-      LogicalJoin(condition=[=($7, $11)], joinType=[inner])
+      LogicalJoin(condition=[=($cor0.DEPTNO, $2)], joinType=[inner])
```

Comment:
Was the previous plan wrong? Is the new plan better?

Reply:
Yes, the previous plan was wrong, and the new plan is better.

This test was previously off and would throw an exception while building the rel nodes.  If you disabled the litimus check, then the original plan is what was generated.

Reply:
@zabetak, Sorry keep bother you but do you need any other changes for PR?

Reply:
@zabetak, any further feedback?  Should I try to get any one else to review it?

Reply:
Plan indeed looks good ;)